be/src/storage/rowset/column_writer.cpp

```diff
@@ -816,4 +871,136 @@ Status StringColumnWriter::check_string_lengths(const Column& column) {
     return Status::OK();
 }
 
+DictColumnWriter::DictColumnWriter(const ColumnWriterOptions& opts, TypeInfoPtr type_info,
+                                   std::unique_ptr<ScalarColumnWriter> column_writer)
+        : ColumnWriter(std::move(type_info), opts.meta->length(), opts.meta->is_nullable()),
+          _scalar_column_writer(std::move(column_writer)) {}
+
+Status DictColumnWriter::append(const Column& column) {
+    if (_is_speculated) {
+        return _scalar_column_writer->append(column);
+    }
+
+    if (_buf_column == nullptr) {
+        // first column size is greater than speculate size
+        if (column.size() >= config::dictionary_speculate_min_chunk_size) {
+            _is_speculated = true;
+            RETURN_IF_ERROR(speculate_column_and_set_encoding(column));
+            return _scalar_column_writer->append(column);
+        } else {
+            _buf_column = column.clone_empty();
+            _buf_column->append(column, 0, column.size());
+            return Status::OK();
+        }
+    }
+    _buf_column->append(column, 0, column.size());
+    if (_buf_column->size() < config::dictionary_speculate_min_chunk_size) {
+        return Status::OK();
+    } else {
+        _is_speculated = true;
+        RETURN_IF_ERROR(speculate_column_and_set_encoding(*_buf_column));
+        Status st = _scalar_column_writer->append(*_buf_column);
+        _buf_column.reset();
+        return st;
+    }
+}
+
+inline Status DictColumnWriter::speculate_column_and_set_encoding(const Column& column) {
+    Status st;
+    EncodingTypePB detect_encoding;
+    LogicalType logicalType = delegate_type(type_info()->type());
+    switch (logicalType) {
+    case TYPE_TINYINT:
+        detect_encoding = speculate_encoding<TYPE_TINYINT>(column);
+        break;
+    case TYPE_SMALLINT:
+        detect_encoding = speculate_encoding<TYPE_SMALLINT>(column);
+        break;
+    case TYPE_INT:
+        detect_encoding = speculate_encoding<TYPE_INT>(column);
+        break;
+    case TYPE_BIGINT:
+        detect_encoding = speculate_encoding<TYPE_BIGINT>(column);
+        break;
+    case TYPE_LARGEINT:
+        detect_encoding = speculate_encoding<TYPE_LARGEINT>(column);
+        break;
+    case TYPE_FLOAT:
+        detect_encoding = speculate_encoding<TYPE_FLOAT>(column);
+        break;
+    case TYPE_DOUBLE:
+        detect_encoding = speculate_encoding<TYPE_DOUBLE>(column);
+        break;
+    case TYPE_DATE:
+        detect_encoding = speculate_encoding<TYPE_DATE>(column);
+        break;
+    case TYPE_DATETIME:
+        detect_encoding = speculate_encoding<TYPE_DATETIME>(column);
+        break;
+    case TYPE_DECIMAL:
+        detect_encoding = speculate_encoding<TYPE_DECIMAL>(column);
+        break;
+    case TYPE_DECIMALV2:
```

Comment:
decimalv2 is deprecated, 
TYPE_DECIMAL32,64,128 should be supported.

Reply:
In fact, decimal_32/64/128 is reprensted by int/bigint/largeint. This pr have supported these types.
![image](https://github.com/StarRocks/starrocks/assets/45813655/7da0595b-fa87-41b7-b985-2735d529e709)

Reply:
@wyb